src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.GenerateGlobalUsings.targets

```diff
@@ -0,0 +1,64 @@
+<!--
+***********************************************************************************************
+Microsoft.NET.GenerateGlobalUsings.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved.
+***********************************************************************************************
+-->
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!--
+    ============================================================
+                                     GenerateGlobalUsings
+
+    Generates global using source to intermediate directory
+    ============================================================
+    -->
+  <PropertyGroup>
+    <GeneratedGlobalUsingsFile Condition="'$(GeneratedGlobalUsingsFile)' ==''">$(IntermediateOutputPath)$(MSBuildProjectName).GlobalUsings$(DefaultLanguageSourceExtension)</GeneratedGlobalUsingsFile>
+    <GenerateGlobalUsings Condition="'$(GenerateGlobalUsings)' == '' and '$(LangVersion)' == 'preview'">true</GenerateGlobalUsings>
+  </PropertyGroup>
+
+  <ItemGroup Condition=" '$(TargetFrameworkIdentifier)' == '.NETCoreApp'
+                      and $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '6.0'))">
+    <GlobalUsings Include="System" />
+    <GlobalUsings Include="System.Collections.Generic" />
+    <GlobalUsings Include="System.Linq" />
+    <GlobalUsings Include="System.Net.Http" />
+    <GlobalUsings Include="System.Net.Http.Json" />
+  </ItemGroup>
+
+  <!--
+    Note that this must run before every invocation of CoreCompile to ensure that all compiler
+    runs see the generated assembly info. There is at least one scenario involving Xaml
+    where CoreCompile is invoked without other potential hooks such as Compile or CoreBuild,
+    etc., so we hook directly on to CoreCompile. Furthermore, we  must run *after*
+    PrepareForBuild to ensure that the intermediate directory has been created.
+
+    Targets that generate Compile items are also expected to run before
+    BeforeCompile targets (common targets convention).
+   -->
+  <Target Name="GenerateGlobalUsings"
```

Comment:
I've been following what GenerateAssemblyInfo does. I think this is not private because the target may be used in Before/AfterTargets and DependsOn since end users may need to modify the GlobalUsing itemgroup and need to control the order.